### PR TITLE
fix: node nqn field not shown in rest api and grpc

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
Signed-off-by: Abhinandan Purkait <purkaitabhinandan@gmail.com>

### CODE REVIEW

File: control-plane/grpc/src/operations/node/traits.rs [hunk 1]
```diff
@@ -78,7 +78,10 @@ impl TryFrom<node::Node> for Node {
                     },
                     None => None,
                 },
-                None,
+                match spec.node_nqn {
+                    Some(node_nqn) => Some(HostNqn::try_from(node_nqn)?),
+                    None => None,
+                },
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/node/traits.rs]:
```suggestion
                spec.node_nqn.try_into_opt()?,
```

File: control-plane/grpc/src/operations/node/traits.rs [hunk 2]
```diff
@@ -104,7 +107,10 @@ impl TryFrom<node::Node> for Node {
                     })?,
                     status,
                     None,
-                    None,
+                    match state.node_nqn {
+                        Some(node_nqn) => Some(HostNqn::try_from(node_nqn)?),
+                        None => None,
+                    },
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/node/traits.rs]:
```suggestion
                state.node_nqn.try_into_opt()?,
```

File: control-plane/grpc/src/operations/node/traits.rs [hunk 3]
```diff
@@ -161,6 +167,7 @@ impl From<Node> for node::Node {
                 }
                 None => None,
             },
+            node_nqn: types_v0_spec.node_nqn().clone().map(|nqn| nqn.to_string()),
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/node/traits.rs]:
```suggestion
            node_nqn: types_v0_spec.node_nqn().map(|nqn| nqn.to_string()),
```

File: control-plane/grpc/src/operations/node/traits.rs [hunk 4]
```diff
@@ -170,6 +177,7 @@ impl From<Node> for node::Node {
                     node_id: types_v0_state.id.to_string(),
                     endpoint: types_v0_state.grpc_endpoint.to_string(),
                     status: grpc_node_status as i32,
+                    node_nqn: types_v0_state.node_nqn.clone().map(|nqn| nqn.to_string()),
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/node/traits.rs]:
```suggestion
                    node_nqn: types_v0_state.node_nqn.map(|nqn| nqn.to_string()),
```

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/src/operations/node/traits.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/node/traits.rs]: Doesn't work without clone.

File: control-plane/grpc/src/operations/node/traits.rs (hunk 4 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/node/traits.rs]: Doesn't work without clone.

**Review by abhilashshetty04 — APPROVED** [review]

File: control-plane/grpc/src/operations/node/traits.rs (hunk 4 above)
Comment (tiagolobocastro) [reply control-plane/grpc/src/operations/node/traits.rs]: Right, it needs .as_ref() instead of .clone()

Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/node/traits.rs]: Yep.

File: control-plane/grpc/src/operations/node/traits.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/node/traits.rs]: used as_ref()

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**